test/testfloat/testfloat.cpp

```diff
@@ -228,13 +228,11 @@ constexpr TypedValue from_bits(ValType ty, uint64_t bits) noexcept
     switch (ty)
     {
     case ValType::i32:
-        return static_cast<uint32_t>(bits);
-    case ValType::i64:
-        return bits;
     case ValType::f32:
-        return FP{static_cast<uint32_t>(bits)}.value;
+        return {ty, Value{static_cast<uint32_t>(bits)}};
```

Comment:
This sets `i32` member of `Value` for `f32` type.

Reply:
This is by design. We don't want to call a function with `float` param (here `Value(float)`) because this converts sNaN to qNaN.

Reply:
Comment added.